File: app/src/main/java/com/pec_acm/moviedroid/mainpage/list/ListFragment.java
package com.pec_acm.moviedroid.mainpage.list;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.widget.ViewPager2;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;
import com.pec_acm.moviedroid.databinding.FragmentListBinding;

public class ListFragment extends Fragment {
    private FragmentListBinding binding;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        binding=FragmentListBinding.inflate(inflater,container,false);
        View view=binding.getRoot();

        ViewPager2 listPager=(ViewPager2) binding.listPager;
        TabLayout listTabLayout=(TabLayout) binding.listTabLayout;

        listPager.setAdapter(new ListPagerAdapter(this));
        new TabLayoutMediator(listTabLayout, listPager, (tab, position) -> {
            String tabText;
            switch (position)
            {
                default:
                    tabText="All";
                    break;
                case 1:
                    tabText="Watching";
                    break;
                case 2:
                    tabText="Completed";
                    break;
                case 3:
                    tabText="On Hold";
                    break;
                case 4:
                    tabText="Dropped";
                    break;
                case 5:
                    tabText="Plan to Watch";
                    break;
            }
            tab.setText(tabText);
        }).attach();

        return view;
    }
}

File: app/src/main/java/com/pec_acm/moviedroid/mainpage/list/ListFragment.kt
package com.pec_acm.moviedroid.mainpage.list

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import com.google.android.material.transition.platform.MaterialFadeThrough
import com.pec_acm.moviedroid.R
import com.pec_acm.moviedroid.databinding.FragmentListBinding

class ListFragment : Fragment() {
    private lateinit var binding: FragmentListBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enterTransition = MaterialFadeThrough()
        exitTransition = MaterialFadeThrough()
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentListBinding.inflate(inflater, container, false)
        val view: View = binding.root
        val listPager = binding.listPager
        listPager.adapter = ListPagerAdapter(this)
        TabLayoutMediator(binding.listTabLayout, listPager) { tab: TabLayout.Tab, position: Int ->
            val tabText: String = when (position) {
                1 -> getString(R.string.watching_tab)
                2 -> getString(R.string.completed_tab)
                3 -> getString(R.string.on_hold_tab)
                4 -> getString(R.string.dropped_tab)
                5 -> getString(R.string.plan_to_watch_tab)
                else -> getString(R.string.all_tab)
            }
            tab.text = tabText
        }.attach()
        return view
    }
}